### DISCUSSION TITLE
Spawing Entities Dynamically

### DISCUSSION THREAD

**Discussion by rdelfin** [Trouv/bevy_ecs_ldtk#384] [Q&A]
Sorry if this is something that's mentioned elsewhere but I couldn't find any issues.

Is there currently a way of manually spawning a new entity based off an entity type in LDtk? To give an example, I have a game where entities spawn based on a specific event from specific spots on the map (the spawn points are themselves entities). What I'd like to do is use the "entity" objects in LDtk as a template to spawn a new entity. That way I can use the spritesheet as loaded in LDtk. Is this possible?

**Comment by TurtIeSocks** [comment]
~~+1 here, also interested. Or would be curious to hear if this is the wrong approach entirely and what would be a better way!~~ I suppose the reasoning for why this isn't possible/apparent are generally covered by https://github.com/Trouv/bevy_ecs_ldtk/issues/258

**Comment by rdelfin** [comment]
Yeah, that's a fair point! Maybe it's easier to have my own custom bundles and logic for doing that in game instead

**Comment by TurtIeSocks** [comment]
> Yeah, that's a fair point! Maybe it's easier to have my own custom bundles and logic for doing that in game instead

Since my comment I did actually manage to make this "work". I'll PR some example code someday but basically I have my bundles for each entity that I treat as templates, spawn them in at startup and then turn their visibility to hidden. Then when I want to spawn them in code, I query for a unique component marker that I've inserted into the template entities, and clone each of the components that I want to the new entity.

**Reply by Trouv** [reply]
Yeah, I'd be happy to look at such a PR, especially now w/ the book it could be a "how-to-guide"

**Comment by Trouv** [comment]
Converted this to a discussion since it's not really a feature request, yet. I don't think this sort of thing would be supported natively by the plugin as the scope is merely "spawning LDtk levels" not "game logic". I could potentially see something like.. a user spawning an entity with a component `EntityDefUid` and the plugin then figuring it out? Maybe that's okay it because it is spawning something pulled from the LDtk assets, but I do think it wouldn't be a decision to make lightly as it is scope creep to not be concerned with mere levels.
